server/utils/referrals/caseListUtils.ts

```diff
@@ -52,6 +61,57 @@ export default class CaseListUtils {
     ])
   }
 
+  static pagination(
+    path: Request['path'],
+    queryParamsExcludingPage: Array<QueryParam>,
+    pageNumber: Paginated<ReferralSummary>['pageNumber'],
+    totalPages: Paginated<ReferralSummary>['totalPages'],
+  ): GovukFrontendPaginationWithItems {
+    const oneIndexedPageNumber = pageNumber + 1
+
+    const pagination: GovukFrontendPaginationWithItems = { items: [] }
+
+    const numberPageLinkNumbers = Array.from(
+      new Set([1, oneIndexedPageNumber - 1, oneIndexedPageNumber, oneIndexedPageNumber + 1, totalPages]),
+    ).filter((number: Paginated<ReferralSummary>['pageNumber']) => number > 0 && number <= totalPages)
+
+    numberPageLinkNumbers.forEach((numberPageLinkNumber, numberPageLinkNumberIndex) => {
+      if (
+        numberPageLinkNumber !== 1 &&
```

Comment:
again, although we'd be declaring variables here and re-using them, I think it'd help readability if we could encapsulate some of this logic in a clear variable name to make this a bit more reader friendly